### ISSUE TITLE
Multi-line textbox

### ISSUE THREAD

**Issue by wesrer** [hecrj/iced#320]
I am fairly new to iced, and just writing some basic code to test the waters. 
I was just wondering if there is any easy way currently to have multi-line text input, because all the example code uses a single line input for `TextInput`.

**Comment by hecrj** [comment]
No, there is no multi-line text input support yet. This is definitely planned, but I believe we will need to tackle #33 and #34 first.

If you want to join the discussion, I shared some of my thoughts in [a Zulip thread](https://iced.zulipchat.com/#narrow/stream/213316-general/topic/more.20text.2Ftext_input.20stuff):

> This will be pretty hard (see [Text Editing Hates You Too](https://lord.io/blog/2019/text-editing-hates-you-too/)).
>
> I believe it should be a new widget with a retained API on top of some interface for text editing ([`ropey`](https://github.com/cessen/ropey) may be worth considering). Proper cursor movement and multi-line text selection will be a challenge, specially if we want to support different scripts.
>
> I feel like we should focus on rendering and layouting text properly first. The idea here would be replacing `wgpu_glyph` with a more GUI-oriented text renderer (i.e. with shaping). There are many parts of the Rust ecosystem that need to mature for this to happen, which means there are many opportunities for contributing!

**Comment by the-drunk-coder** [comment]
I'm at the same point, looking for an easy, pure rust text editor widget ... my attempt to modify the current one wasn't fruitful so far ... I got multiple lines but everything else was off :sweat_smile:

**Comment by dhardy** [comment]
> easy, pure rust text editor widget 

If that's all you want, both [KAS](https://github.com/kas-gui/kas/tree/master/kas-wgpu/examples#markdown) and [Druid](https://github.com/linebender/druid/blob/master/druid/examples/styled_text.rs) have multi-line text boxes. Don't expect them to be perfect and I'm not sure about easy. Text is insanely complicated once you start digging into the subject.

**Comment by Dimev** [comment]
How far along is at the moment?

**Comment by 13r0ck** [comment]
@Dimev not sure, but I recommend that you be VERY patient.

**Comment by benluelo** [comment]
Hello, is there any progress on this issue? I would love to contribute to whatever needs help to get this issue moving. I see that #33 and #34 are mentioned in an above comment, where should I start?

**Comment by CodeDead** [comment]
Definitely a nice-to-have!

**Comment by jellybobbin** [comment]
Definitely a nice-to-have!

**Comment by henry-lang** [comment]
would appreciate

**Comment by adracea** [comment]
Hello, is there any status update for this? I'm seeing the two mentioned PRs are coming along but no idea how far away this would be.

**Comment by jquesada2016** [comment]
I see #33 was closed, so not sure where this currently stands?

**Comment by thunderstorm010** [comment]
According to the [roadmap](https://whimsical.com/roadmap-iced-7vhq6R35Lp3TmYH4WeYwLM) it is planned for 0.12.

**Comment by max397574** [comment]
couldn't this be closed now?

**Comment by hecrj** [comment]
Implemented in #2123.
